### PR TITLE
Fix batch processing bug and modify delay values.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
- fix a bug causing an infinite loop.
- modify delay values so that timeline data processes faster and cpu profile data processes slightly slower (allows the progress indicator to appear more smooth).

### CODE REVIEW

File: packages/devtools_app/lib/src/profiler/cpu_profile_transformer.dart [hunk 1]
```diff
@@ -54,8 +54,9 @@ class CpuProfileTransformer {
       _progressNotifier.value = _stackFramesProcessed / _stackFramesCount;
 
       // Await a small delay to give the UI thread a chance to update the
-      // progress indicator.
-      await delayForBatchProcessing();
+      // progress indicator. Use a longer delay than the default (0) so that the
+      // progress indicator will look more smooth.
```
Comment (terrylucas) [inline packages/devtools_app/lib/src/profiler/cpu_profile_transformer.dart]: nit: smoother

File: packages/devtools_app/lib/src/profiler/cpu_profile_transformer.dart [hunk 2]
```diff
@@ -54,8 +54,9 @@ class CpuProfileTransformer {
...
+      await delayForBatchProcessing(micros: 5000);
```
Comment (terrylucas) [inline packages/devtools_app/lib/src/profiler/cpu_profile_transformer.dart]: is it milliseconds not microseconds?

File: packages/devtools_app/lib/src/timeline/timeline_processor.dart [hunk 3]
```diff
@@ -603,7 +603,7 @@ class FullTimelineProcessor extends TimelineProcessor {
       : super(timelineController);
 
   /// Number of traceEvents we will process in each batch.
-  static const _defaultBatchSize = 250;
+  static const _defaultBatchSize = 2000;
```
Comment (terrylucas) [inline packages/devtools_app/lib/src/timeline/timeline_processor.dart]: Should this number be based on the delay for batchProcessing or does it not matter?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/timeline_processor.dart]: The delay is more important for the appearance of the progress indicator. This number should be whatever makes sense for the type of data being processed. The number of trace events we can be in the tens of thousands, so we want the data to be able to process quickly. I manually tested a release build for several different batch sizes and delays, and this is what I settled on to provide quick processing and a smooth progress indicator. With the previous size of 250, it added ~10 seconds of processing time to a typical trace from dream (g3), which is how I discovered that these numbers needed to be modified.

**Pull request merged**